gregBorchers/Homework2GregBorchers-probs1-4NoCards.js

```diff
@@ -0,0 +1,359 @@
+                    ,"On the eighth day of Christmas, my true love gave to me, "
+                    ,"On the ninth day of Christmas, my true love gave to me, "
+                    ,"On the tenth day of Christmas, my true love gave to me, "
+                    ,"On the eleventh day of Christmas, my true love gave to me, "
+                    ,"On the twelfth day of Christmas, my true love gave to me, "];
+
+// TODO - improvements would be to use looping arrays to reduce the static storage of repeated strings.
+// "my true love gave to me" etc. could be concatenated, etc.
+
+   //  loop through the 12 Days, singing each the new thing (index) and down (index to 0) the lists
+   var songString = [""];
+   for ( var i = 0; i < twelveDaysList.length; i++) 
+    { // counting up the days list
+      songString += twelveDaysList[i] + "\n";
+      for (var j = i; j >= 0; j--)  // counting up the gift list to the current day index
+        {
+          if ( i > 0 && j ===0 )
+          {
+            songString += ("and " + twelveGiftsList[j] + "\n" );
+          } 
+          else
+          {
+            songString += (twelveGiftsList[j] + "\n" );
+          }
+      }
+   }
+   console.log(songString);
+}
+
+singTwelveDays();
+
+
+/*
+Lyrics for the song: reference: http://en.wikipedia.org/wiki/The_Twelve_Days_of_Christmas_%28song%29
+On the First day of Christmas my true love sent to me
+a Partridge in a Pear Tree.
+
+On the Second day of Christmas my true love sent to me
+Two Turtle Doves
+and a Partridge in a Pear Tree.
+
+On the Third day of Christmas my true love sent to me
+Three French Hens,[3]
+Two Turtle Doves
+and a Partridge in a Pear Tree.
+
+Subsequent verses follow the same pattern, each adding one new gift and repeating all the earlier gifts, so that each verse is one line longer than its predecessor:
+a Partridge in a Pear Tree.
+Two Turtle Doves
+Three French Hens
+Four Calling Birds
+Five Gold Rings
+Six Geese a-Laying
+Seven Swans a-Swimming
+Eight Maids a-MilEight
+Nine Ladies Dancing
+Ten Lords a-Leaping
+Eleven Pipers Piping
+Twelve Drummers Drumming
+*/
+
+
+// If you prefer a non-Christmas option, you may choose a different song with similarly repeating structure, such as "There was an Old Woman Who Swallowed a Fly".  If you prefer a vegan option, you may write your own cruelty-free song.
+
+// ---
+
+// **4) _(Difficult, 20%)_
+
+// Suppose the '&' key on your keyboard is missing, and you want to be able to
+// continue programming without it.  You decide to simulate the && operator with a
+// function!
+
+
+
+// **a)** Write a function `and2(a,b)` which tries to simulate the && operator: it
+// should always return the same result as `(a && b)` for any values of _a_ and
+// _b_.  (For example, `and2((0>1),true)` should return _false_.)  But you can't
+// use && itself within your function!
+
+var and2 = function(a,b){
+  return (a*b ? true : false); 
+};
```

Comment:
4) Remember that the `&&` operator is a logical AND: it checks to see if both things passed to it are true. Multiplication doesn't really enter into it, as you can't multiply a string (which can be true) by a number (which can be true)--you'll get `NaN` if you try, which is a false-ish value.